### PR TITLE
Add platform developer access policy

### PR DESCRIPTION
**Pull request by benbc**

### CODE REVIEW

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:50]:
```suggestion
Note: this section applies to OpenSAFELY-TPP and OpenSAFELY-EMIS only.
```

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:100]: nit: This might be slightly easier to follow as a bulleted list:

```suggestion
Data curation activities may be carried out:

* when new datasets are added to OpenSAFELY
* when datasets are updated
* periodically for data quality assurance and sense checking purposes
```

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:96]: question: Do we need to define these roles somewhere — possibly the glossary? — if they are not defined already?

I don't remember ever seeing "Data Investigator". From a search, "core" is used in a couple of ambiguous ways in the documentation; for example, "core OpenSAFELY team".

File: docs/developer-access-policy.md
Comment (iaindillingham) [inline docs/developer-access-policy.md:48]: Nit: Inconsistent italicization. IMHO we don't need to italicize it, because it's a word that's now entered the English language and is no longer on loan from Latin.

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:35]:
```suggestion
including:

* publication of all code that is run
* logging of all operations
* manual checking of outputs for disclosivity before they are released from the backend

```

File: docs/developer-access-policy.md
Comment (iaindillingham) [inline docs/developer-access-policy.md:8]: Nit: We could remove "raw" here, IMHO. We don't use it in the first sentence (above) and, thinking defensively, "raw" might seem to contradict "pseudonymized".

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:25]: suggestion:

The link pointed to has a heading of "project pipelines". But the link text is "study pipelines". I'd personally go for consistency and use "project pipelines" here. [^1]

[^1]: If the correct term really is "study pipelines", we should change that everywhere later.

Comment (StevenMaude) [reply docs/developer-access-policy.md:25]: comment:

This sentence sounds like it is now talking about both normal running — "service provision" — of OpenSAFELY and testing. Perhaps it's simpler to solely focus on the "software maintenance" aspect? That is what is being discussed. This means you can rewrite as:

> For software maintenance purposes, it is sometimes necessary to run OpenSAFELY project pipelines.

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:26]:
```suggestion
The test project pipelines are necessary to:

* ensure that the entire OpenSAFELY platform is operating correctly
* verify that changes to the software work correctly in deployed systems
```

File: docs/developer-access-policy.md
Comment (iaindillingham) [inline docs/developer-access-policy.md:64]: Nit: This is a case of the text reading better in source than in rendered versions. This is an inline list, where each list item begins "to...". Possibly separate list items with semicolons rather than commas?

```
This process gives access
to the OpenSAFELY software running in the backend and to logs that that software produces;
to pseudonymized patient data;
to unvetted intermediate data;
and to unchecked, unreleased study outputs.
```

**Review by iaindillingham — APPROVED** [review]
A couple of nitpick comments, but none block merging.

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:64]:
```suggestion
This process gives access to:

* the OpenSAFELY software running in the backend
* logs that the OpenSAFELY software produces
* pseudonymized patient data
* unvetted intermediate data
* unchecked, unreleased study outputs.
```

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:79]:
```suggestion
!!! note
    This section applies to OpenSAFELY-TPP and OpenSafely-EMIS only.
```

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:9]:
```suggestion
A small number of OpenSAFELY team members need more direct access to the raw pseudonymised data inside the secure environment for:

* developing and running the OpenSAFELY platform
* data curation purposes 
```

I've also switched these two points around to match the document order.

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:48]:
```suggestion
## Ad hoc access to backend systems
```

I'd remove the italics for consistency with the usage above. From reading around, it tends not to be italicised.

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:55]:
```suggestion
To ensure the correct operation of the platform,
it is sometimes necessary to directly access OpenSAFELY backend systems so that:

* software can be reconfigured or updated
* problems which only manifest themselves in the deployed system can be investigated
```

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:46]: question: Is this a necessary detail? It might be, it might not. The previous two sentences read fine to me without it.

I'd take it as given that we aim to maintain the security promises of the platform. But maybe it does need to be stated?

Comment (StevenMaude) [reply docs/developer-access-policy.md:46]: If it's necessary, I'd replace the semi-colon on previous line with a full stop.

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:66]:
```suggestion
There is no facility for extracting data from the system via this process: the data can only be studied within the system itself.
```

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:83]: nit: This is slightly tricky to read for me: is "sense-checking" part of "checking data quality"? "sense-checking" might also be a little ambiguous.

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:92]:
```suggestion
including:

* publication of all code that is run
* logging of all operations
* manual checking of outputs for disclosivity before they are released from the backend
```

File: docs/developer-access-policy.md
Comment (StevenMaude) [inline docs/developer-access-policy.md:19]:
```suggestion
### What this policy describes

This policy describes:

* the conditions under which individuals with these roles may access the OpenSAFELY platform
* the controls in place to ensure that such access cannot be used to subvert the OpenSAFELY security model

All such access conforms to the OpenSAFELY platform’s principles of security, transparency and minimal access.

The specific types of access activities this policy covers are:

* Running OpenSAFELY test pipelines
* Ad hoc access to backend systems
* Data curation activities
```

**Review by StevenMaude — APPROVED** [review]
This looks fine. I've added some small editorial suggestions for clarity and consistency [^1].

I like the "why? what? who? when?" structure.

[^1]: In some cases, these are near duplicates of @iaindillingham's, who happened to be reading at the same time.

File: docs/developer-access-policy.md
Comment (benbc) [reply docs/developer-access-policy.md:96]: They are roles defined in the job server. There is an [outstanding issue](https://github.com/opensafely-core/job-server/issues/1624) to document them completely. And once this PR is merged I'll add a note to that issue to put hyperlinks from this policy to the definitions.

File: docs/developer-access-policy.md
Comment (benbc) [reply docs/developer-access-policy.md:59]: I think the detail is helpful. Since this is (at least partly) for testing purposes, it's good to reassure people that we're not just throwing any old code up there, but code that has already been validated. I'll split the sentence.

File: docs/developer-access-policy.md
Comment (benbc) [reply docs/developer-access-policy.md:102]: I've radically simplified this since most of it is duplicated in the "when" section below.

**Comment by benbc** [comment]
@StevenMaude and @iaindillingham thank you both for your careful reviews. The suggested changes make this kind of thing infinitely easier.

**Comment by iaindillingham** [comment]
The `suggestion` block is great. I didn't notice it until I saw @StevenMaude's comments. Thanks Steve!

**Pull request merged**
